blockend/public-kernel.js

```diff
@@ -89,9 +118,27 @@ function TinyBoard (size = 50, ttl = ONE_HOUR, now = Date.now) {
           if (expiresAt < now()) return 'RantExpired'
         } break
 
-        case TYPE_BUMP:
+        case TYPE_BUMP:{
           // TODO: model stun-lock with diminishing returns
-          break
+          const rant = state[btok(CHAIN)] // rant
+
+          // get who is bumping
+          const hasBumped = rant.bumpedBy.some((key) => btok(key) === btok(block.key))
+          if (hasBumped) return 'AlreadyBumped'
+
+          if (data.date > now()) return 'TimeManipulation'
+
+          if (data.date > rant.expiresAt) return 'TooLate'
+
+          if (data.date < rant.lastBumpAt) return 'TimeManipulation'
+
+          const timeSinceLastBump = (now() - rant.lastBumpAt)
+          if (timeSinceLastBump < 1000) return 'TooSoonToBump'
```

Comment:
Nice, I didn't think of this, maybe even raise the number to 10-30seconds to give the new chain state some more time to propagate through the network. it's a very useful mechanic to slow down bump-spam/unwarranted bumps.

Reply:
10sec should be a good number